h2/src/main/org/h2/value/DataType.java

```diff
@@ -898,7 +898,7 @@ public static String getTypeClassName(int type, boolean forResultSet) {
         case Value.RESULT_SET:
             return ResultSet.class.getName();
         case Value.GEOMETRY:
-            return GEOMETRY_CLASS_NAME;
+            return GEOMETRY_CLASS != null ? GEOMETRY_CLASS_NAME : String.class.getName();
```

Comment:
this seems dubious, it is going to cause weird failure modes for people depending on whether or not they have the right things in their classpath. I would say we should rather just throw an exception if they ask for geometry in a format we don't support

Reply:
It only reports what H2 will return in case of `ResultSet.getObject()` invocation. With these changes a JTS `Geometry` is still returned in presence of JTS library, and a `String` is returned if this library is not found.

`byte[]` (EWKB), `String` (EWKT), and JTS `Geometry` are all accepted by H2 in setObject() / updateObject() etc methods.

Also a specific class can be explicitly requested with `getObject(column, Geometry.class)`. 

I don't think that `ResultSetMetaData.getColumnClassName()` method should throw an exception in this case, it will be a violation of its specification.

With these changes there is no dependency on JTS any more for `GEOMETRY` data type, it's fully operable without it.

if JTS `Geometry` data type is used in application for `Geometry g = (Geometry) resultSet.getObject(2)` absence of JTS in classpath will cause self-explaining error here anyway.

If JTS `Geometry` is not used in application for some reason, why we should require it to read metadata of a result set with `GEOMETRY`?

Reply:
ok